### DISCUSSION TITLE
Running tests locallly

### DISCUSSION THREAD

**Discussion by qrazi** [studioespresso/craft-scout#290] [Q&A] [answered]
In previous versions the tests would run on the local environment with a single Docker container for the database. This was complementary to the GH Actions test flows.

How do contributors test now? Is this now a "commit, push, see results, amend commit, push"-flow?

**Comment by janhenckens** [comment, marked as answer]
Hey @qrazi, it wasn't my intention to prevent you from running them through Docker. That's not how I work(ed) and I didn't think removing the file would hinder anyone - I was wrong ;)

Added it again!

**Reply by qrazi** [reply]
Thanks! I am just wondering how to test locally without Docker though; I feel like I should be the one adjusting to your conventions ':)

I did try to get GH Action running locally with https://github.com/nektos/act, but haven't got that to run successfully.

**Reply by janhenckens** [reply]
I'm either running it through Docker, or I create a second database in [ddev](https://ddev.readthedocs.io/en/stable/) and point my test to that. I've also tried my luck at getting `act` running but without success.

What sort of environment are you using for local development?

**Reply by qrazi** [reply]
Docker as well, RYO using Traefik and mkcert for *.test domains etc. So, kind of ddev / sail... :sweat_smile: 

For this repo specifically I am running it on native PHP install (+ the Docker db). 

If you want you can remove the Dockerfile again, in retrospect it also makes a lot of sense to leave the test infrastructure completely up to the develop :sweat_smile:

**Reply by janhenckens** [reply]
I'm fine keeping it, in my current setup it's actually easier to run them through docker. And I plan on giving the tests a full review with the Craft 5 upgrade (including some docs on how to test) so it's good to have a set way of doing things.
